backend/fms_core/services/derived_sample.py

```diff
@@ -18,3 +19,18 @@ def inherit_derived_sample(derived_sample_source, new_derived_sample_data):
 
     return (new_derived_sample, errors, warnings)
 
+def get_library_size_for_derived_sample(derived_sample_id: int) -> Optional[int]:
+    """
+    Provides the latest measured fragment_size related to a derived_sample
+
+    Args:
+        `derived_sample_id`: Derived_sample for which we want the latest measured fragment_size
+
+    Returns:
+        An integer that represents the fragment_size (library_size), None if not found or never measured.
+    """
+    samples_with_library_size = DerivedBySample.objects.filter(derived_sample_id=derived_sample_id, sample__fragment_size__isnull=False)
+     # Most recent sample in the lineage chain will have a larger id
+    ordered_samples_with_library_size = samples_with_library_size.order_by("-sample__parent_sample__id")
+    library_size = ordered_samples_with_library_size.values_list("sample__fragment_size", flat=True).first()
```

Comment:
oh no wait, the function should be returning a single value

Reply:
Ideally, at some point, you should index the sequence maybe with `[0]` to make it explicit that it's one value.
Oh but wait, Django returns a value if it's just one value. Django being weird.

Reply:
the key part here is the .first() . It either returns one item if there is one but it still returns None if there is not any selected (instead of failing as would [0]).

Reply:
oh woops, missed that part